app/main/routes.py

```diff
@@ -82,6 +82,8 @@ def callback():
 
     session["access_token"] = access_token_response["access_token"]
     session["refresh_token"] = access_token_response["refresh_token"]
+    decoded_access_token = keycloak_openid.introspect(session["access_token"])
+    session["user_groups"] = decoded_access_token["groups"]
```

Comment:
noticed that neither sign-in nor callback routes are covered by normal pytest flask tests, only by e2e playwright tests. The fact that all our permission based browse and search tests and the sign in tests work in e2e is a good coverage already but I may add a couple of simple tests for these routes in a followup PR which mock keycloack and assert on the session.

Reply:
I agree, we should have at least basic tests for code coverage for sign-in and callback routes